Suppress deprecation message of E_STRICT constant

MauricioFauth opened the pull request:
The `E_STRICT` constant was deprecated in PHP 8.4.

It's still a possible error level when using PHP 7.2 and PHP 7.3.

See: https://wiki.php.net/rfc/deprecations_php_8_4#remove_e_strict_error_level_and_deprecate_e_strict_constant

kamil-tekiela commented on libraries/classes/Error.php:
```diff
@@ -316,7 +312,11 @@ public function getLine(): int
     public function getType(): string
     {
-        return self::$errortype[$this->getNumber()];
+        if ($this->getNumber() === @E_STRICT) {
```
Why not keep it in the array and just use a number instead with a comment beside what the number represents?

MauricioFauth replied:
I don't like to use the numbers directly, but I think it's better in this case, as `E_STRICT` should be removed in `master` branch.

kamil-tekiela commented on libraries/classes/ErrorHandler.php:
```diff
@@ -281,7 +281,7 @@ public function addError(
 
         switch ($error->getNumber()) {
-            case E_STRICT:
+            case @E_STRICT:
```
```suggestion
            case 2048; // E_STRICT
```

MauricioFauth committed:
Suppress deprecation message of E_STRICT constant

The E_STRICT constant was deprecated in PHP 8.4.

It's still a possible error level when using PHP 7.2 and PHP 7.3.

See:
https://wiki.php.net/rfc/deprecations_php_8_4#remove_e_strict_error_level_and_deprecate_e_strict_constant

Signed-off-by: Maurício Meneghini Fauth <mauricio@mfauth.net>

kamil-tekiela commented:
Out of curiosity, if it's only a deprecation why does it go to 5.2 branch? The code still works, right?

MauricioFauth commented:
> Out of curiosity, if it's only a deprecation why does it go to 5.2 branch? The code still works, right?

Yes, it still works.

But in this case, when in development environment, as the deprecation is triggered in the ErrorHandler when handling an exception, the deprecation is converted to an ErrorException and it won't be caught.

Also, besides deprecation messages not being shown to users, I think it's worth "fixing" it if it's possible.
